### PR TITLE
Support for UPSERT ... UPDATE

### PR DESCRIPTION
**Pull request by venkatd**
This is just a proof of concept. I adopted the logic from the Postgres adapter. Tests have not been updated but it appears to work with a local import script I am using that makes use of UPSERT.

#225

### CODE REVIEW

**Comment by ConnorRigby** [comment]
I will see what i can do about getting a `esqlite` release published, then  merge this. :+1: Thanks @venkatd

**Comment by venkatd** [comment]
Great! I'm in no rush as I can use our fork in the meantime.

File: mix.exs [hunk 1]
```diff
@@ -46,6 +46,7 @@ defmodule Sqlite.Ecto2.Mixfile do
      {:poison, "~> 2.2 or ~> 3.0", optional: true},
      {:postgrex, "~> 0.13", optional: true},
      {:sbroker, "~> 1.0"},
+     {:esqlite, github: "mmzeeman/esqlite", branch: "master", override: true},
```
Comment (ConnorRigby) [inline mix.exs]: I published updated `esqlite`. This can be removed

File: mix.exs [hunk 2]
```diff
@@ -46,6 +46,7 @@ defmodule Sqlite.Ecto2.Mixfile do
...
      {:sqlitex, "~> 1.4"}]
```
Comment (ConnorRigby) [inline mix.exs]: This can be updated to `{:sqlitex, "1.5.0"}`

File: test/sqlite_ecto_test.exs [hunk 3]
```diff
@@ -753,39 +753,30 @@ defmodule Sqlite.Ecto2.Test do
...
+    update = normalize(from("schema", update: [set: [z: "foo"]]), :update_all)
+    query = insert(nil, "schema", [:x, :y], [[:x, :y]], {update, [], [:x, :y]}, [:z])
+    assert query = ~s{INSERT INTO "schema" ("x","y") VALUES (?1,?2) ON CONFLICT ("x","y") DO UPDATE SET "z" = 'foo' ;--RETURNING ON INSERT "schema","z"}
 
-    # For :replace_all
-    assert_raise ArgumentError, "Upsert in SQLite must use on_conflict: :nothing", fn ->
-      insert(nil, "schema", [:x, :y], [[:x, :y]], {:replace_all, [], [:id]}, [])
-    end
+    update = normalize(from("schema", update: [set: [z: ^"foo"]], where: [w: true]), :update_all, 2)
+    insert(nil, "schema", [:x, :y], [[:x, :y]], {update, [], [:x, :y]}, [:z]) |> IO.puts
```
Comment (ConnorRigby) [inline test/sqlite_ecto_test.exs]: Can you remove the `IO.puts` here?

**Comment by venkatd** [comment]
@ConnorRigby sorry for having gone MIA. This month has been crazy. I'll be on vacation until mid-Jan. Once I return, I'll get this sorted out!

**Comment by ConnorRigby** [comment]
Closed by #233

**Pull request closed without merging**